File: CompilationTest/StrLenInferIssue/results/StrLenInferIssue.cpp
#include <iostream>
#include <string>

class StrLenInferIssue {
  public:
    static int test(std::string str) {
        return str.size();
    }

  private:
};

int main()
{
    std::cout << StrLenInferIssue::test(std::string("hello")) << std::endl;
    return 0;
}

File: CompilationTest/StrLenInferIssue/results/StrLenInferIssue.pl
use strict;
use warnings;

package StrLenInferIssue;

sub new
{
    my $class = shift;
    my $self = {};
    bless $self, $class;
    return $self;
}

sub test {
    my ( $str ) = @_;
    return length($str);
}

print((StrLenInferIssue::test("hello")) . "\n");